inference/explained_bool.go

```diff
@@ -113,44 +140,48 @@ func (f FalseBecauseDeepConstraint) String() string {
 		f.InternalAssertion.ProducerRepr, f.InternalAssertion.ConsumerRepr, f.DeeperExplanation.String())
 }
 
+func (f FalseBecauseDeepConstraint) getPrimitiveFullTrigger() primitiveFullTrigger {
+	return f.InternalAssertion
+}
+
+func (f FalseBecauseDeepConstraint) getExplainedBool() ExplainedBool {
+	return f.DeeperExplanation
+}
+
 // TrueBecauseAnnotation is used as the label for a site X on which a literal annotation "//nilable(x)"
 // has been discovered - forcing that site to be nilable.
 type TrueBecauseAnnotation struct {
 	ExplainedTrue
+	Pos token.Pos
 }
 
 func (TrueBecauseAnnotation) String() string {
 	return "NILABLE because it is annotated as so"
 }
 
+func (t TrueBecauseAnnotation) getPrimitiveFullTrigger() primitiveFullTrigger {
+	return primitiveFullTrigger{Pos: t.Pos}
+}
+
+func (t TrueBecauseAnnotation) getExplainedBool() ExplainedBool {
+	return nil
+}
+
 // FalseBecauseAnnotation is used as the label for a site X on which a literal annotation "//nonnil(x)"
 // has been discovered - forcing that site to be nonnil.
 type FalseBecauseAnnotation struct {
 	ExplainedFalse
+	Pos token.Pos
 }
 
 func (FalseBecauseAnnotation) String() string {
 	return "NONNIL because it is annotated as so"
 }
 
-// TrueBecauseMyopia is used as the label for a site X that a myopic inference procedure decided should
-// be nilable
-type TrueBecauseMyopia struct {
-	ExplainedTrue
-	PkgName string
-}
-
-func (t TrueBecauseMyopia) String() string {
-	return fmt.Sprintf("NILABLE because myopic annotation inference for package %s decided so", t.PkgName)
-}
-
-// FalseBecauseMyopia is used as the label for a site X that a myopic inference procedure decided should
-// be nonnil
-type FalseBecauseMyopia struct {
-	ExplainedFalse
-	PkgName string
+func (f FalseBecauseAnnotation) getPrimitiveFullTrigger() primitiveFullTrigger {
+	return primitiveFullTrigger{Pos: f.Pos}
```

Comment:
Weird that this line isn't reached by tests, actually (not something to fix in this PR but it strikes me as curious. When is `FalseBecauseAnnotation` used?)

Reply:
Hmm.. `FalseBecauseAnnotation` is used when a program site is explicitly annotated as `nonnil`. I can see a couple of instances in our test cases, where this is present. So, not sure why isn't it covered by tests. I'll look into it.